### ISSUE TITLE
Bug: [consistent-type-imports] False negatives when type name also refers to a local value

### ISSUE THREAD

**Issue by haskellcamargo** [typescript-eslint/typescript-eslint#8315]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=PTAEAEBcGcFoFMAeAHeBjSCBOWD2XRZQAVAC3lAFtcATAVwBsKbd5oA7AcklCQEtokADSgARnR6RSA0DPa4eWeEwBuAQ3aQAdACg%2BlZPh4BvUAAUGatPFK4GNeAQC%2BoAGZ5KoTskvXb9x04Abh0dJEMsSQBPVFAAYVx2Vz4Ac1AAXlBjHVBQHysbOwcsAC5zX0KArBCnUJBQAEFQaCjKUTtQAHc%2BKVApCmg1Sgp2IYo1aD7yPpj4HVc6dgw%2BRPKC-2KACgBKLJzQJUg6LHZQdkYGGrqwAFVoPnY0-tAGXDQ1BmbW9s%2BJ0DVQOoGHQ5uEjG5FstVgBZNQPHZ7XKHY6nAA8FnWRUcoGAAD4rkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyoMkRNGhTI4MAF8QaoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
// @ts-expect-error - The module doesn't exist, but this is not relevant.
import { Placeholder } from 'placeholder';

export type Config = {
  // Using it only as a type
  placeholder: Placeholder;
}

// A symbol with the same name as the type
function Placeholder() {
  return null;
}

// Using the local symbol as a value
export function Main() {
  return <Placeholder />;
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/consistent-type-imports": "error"
  }
}
```

### Expected Result

It should report an error because the actual import is only used as a type. Instead no error is reported.
TSC accepts types with the same name as local values. Babel with the TypeScript preset throws an error due to having a duplicate identifier as it is unable to determine that the import will be dropped during compile-time.
There's also changes coming to [TypeScript 5.4](https://devblogs.microsoft.com/typescript/announcing-typescript-5-4-beta/) that introduces errors when type-only imports conflict with local values.

### Actual Result

No errors shown.

**Comment by bradzacher** [comment]
> There's also changes coming to TypeScript 5.4 that introduces errors when type-only imports conflict with local values.

This is as good a reason as any to build this in now so people can autofix their codebase before 5.4!

**Comment by nayounsang** [comment]
Is this issue solved by https://github.com/typescript-eslint/typescript-eslint/pull/11322 ??
